BinScatter/R/BinScatter.R

```diff
@@ -0,0 +1,258 @@
+      tempBinType <- "explicit"
+    } else if (binType == "quantile") {
+      tempBinType <- "quantile"
+    } else {
+      stop(sprintf("binType for %s must be uniform or quantile", var))
+    }
+    
+    # Create a data frame of bin indicators
+    vals           <- data[, eval(var)] %>% as.numeric()
+    bins           <- bin_data(vals, bins = nBins, binType = tempBinType) %>% factor(ordered = FALSE)
+    df_bins        <- data.table(bins) %>% one_hot()
+    names(df_bins) <- paste0(var, "_bin_", c(1:nBins))
+    
+    if (intercept) {
+      df_bins        <- df_bins[, 2:nBins]
+      names(df_bins) <- paste0(var, "_bin_", c(2:nBins))
+    }
+    
+    # Return additionally bin centers for the `x_var`
+    if (is_xvar) {
+      if (binType == "uniform") {
+        bin_center <- sapply(unique(bins), midpoint) %>% sort()
+      } else if (binType == "quantile") {
+        bin_center <- sapply(unique(bins), meanval, vals) %>% sort()
+      }
+      return(list(indicator = df_bins, center = bin_center))
+    } else {
+      return(df_bins)
+    }
+  }
+
+  
+  ##------------------------------------------------------------------------------------------------
+  ## Prepare data for regression
+  ##------------------------------------------------------------------------------------------------
+  
+  ## Clean data
+  data <- data %>% as.data.frame()
+  data <- data[, all_of(c(y_var, x_var, linpartial_var, binpartial_var))]
+  nRaw <- nrow(data)
+  data <- data %>% drop_na() 
+  nNA  <- nRaw - nrow(data)
+  if (nNA > 0) {
+    if (dropNA) { 
+      warning(sprintf("%s observations dropped due to missing value.", nNA)) 
+    } else {
+      stop("Missing values present. Set dropNA == TRUE to overide")
+    }
+  }
+  
+  df_reg <- data %>% dplyr::select(eval(y_var))
+  
+  ## Create bin indicators for `x_var`
+  xbins  <- make_bin_indicator(x_var, data, nBins, binType, intercept, is_xvar = TRUE)
+  x_vals <- xbins[["center"]]
+  df_reg <- cbind(df_reg, xbins[["indicator"]])
+  
+  ## Create bin indicators for variable(s) in `binpartial_var`
+  if (!is.null(binpartial_var)) {
+    
+    if (is.null(nPartialBins))   { nPartialBins <- nBins }
+    if (is.null(partialBinType)) { partialBinType <- binType}
+    
+    df_reg <- cbind(df_reg, lapply(c(binpartial_var), make_bin_indicator, data, nPartialBins, 
+                                   partialBinType, intercept) %>% reduce(cbind))
+  }
+  df_reg <- cbind(df_reg, data[, all_of(c(linpartial_var))])
+  df_reg <- df_reg %>% drop_na()
+  
+  
+  ##------------------------------------------------------------------------------------------------
+  ## Calculate partialed means (and std. err.) for `y_var` (and recenter)
+  ##------------------------------------------------------------------------------------------------
+  
+  model <- sprintf("%s ~ %s - 1", y_var, paste0(names(df_reg)[-1], collapse = " + "))
+  coef <- names(df_reg)[2:(1+nBins)]
+  y_mean <- c()
+  y_se <- c()
```

Comment:
For code clarity, maybe these lines should be in an `else` clause for the `if` statement below, since this applies only when `intercept == FALSE`.

Reply:
Fixed!